### PR TITLE
Custom rule of max-len and no-use-before-define

### PR DESCRIPTION
**Pull request by terryyu**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -11,5 +11,16 @@ module.exports = {
...
+      'code': 120,
+      // sometimes we are providing a json response sample in the comment which is too long
+      'ignoreComments': true,
+      // certain url could be pretty long
+      'ignoreStrings': true,
+      // similar to the string above
+      'ignoreTemplateLiterals': true
+    }],
+    // to allow helper functions to be defined below the core function to get an overview of the feature quickly
+    'no-use-before-define': [ERROR, { 'functions': false }],
```
Comment (minhtule) [inline .eslintrc.js]: Not sure if I understand this rule. I think we can define a helper function below the core function as long as we only call the helper function inside the core function. Can you give an example when we need this?

Comment (terryyu) [reply .eslintrc.js]: an example could be a file that is exporting a function to make cakes. it's much easier to read the assemble function at the top to get the overview of the function, then if you are interested in specific procedure, you can easily navigate from the top. It's similar to you provide an outline of a long articles at the top of the file.

```javascript
export default function makeCake() {
   prepareIngredients();
   prepareOven();
   bake();
   package();
   ...more helpers
}

function  prepareIngredients() {
...
}

function prepareOven() {
...
}

function bake() {
...
}

function package() {
...
}

moreHelpers...
```

**Comment by terryyu** [comment]
> If we want to add these rules to the `eslint-config-ebates` lib, we need to put these rules in the `index.js` file. This file is only for linting this repo...

lol. thanks for pointing out, will close and create a new PR

File: .eslintrc.js (hunk 1 above)
Comment (minhtule) [reply .eslintrc.js]: Seems like I've mistaken with the React components (which can be used before declared in an outers cope) 😬

**Pull request closed without merging**